initramfs/src/main.rs

```diff
@@ -52,7 +52,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args = Args::parse();
     debug!("Running cli with arguments : {:?}", args);
 
-    let registry = Registry::new(&args.registry_url, &args.auth_url);
+    let mut registry = Registry::new(&args.registry_url, &args.auth_url);
```

Comment:
I don't see the point of this mut, is it really useful?

Reply:
`registry.getImage` (called a few lines below), needs a `&mut self`